cli/cmd/package_manifest.go

```diff
@@ -252,6 +263,45 @@ func (c *cliState) detectActiveKernel() (string, bool) {
 	return strings.TrimSuffix(string(kernel), "\n"), true
 }
 
+func (c *cliState) IsEsmEnabled() bool {
+	type uaStatusFile struct {
+		SchemaVersion string `json:"_schema_version,omitempty"`
+		Status        string `json:"execution_status,omitempty"`
+		Services      []struct {
+			Name   string `json:"name,omitempty"`
+			Status string `json:"status,omitempty"`
+		} `json:"services,omitempty"`
+	}
+
+	if file.FileExists(procUAStatusFile) {
+		c.Log.Debugw("detecting ubuntu ESM support", "file", procUAStatusFile)
+		uaStatusBytes, err := ioutil.ReadFile(procUAStatusFile)
+		if err != nil {
+			c.Log.Warnw("unable to read UA status file", "error", err)
+			return false
+		}
+
+		var uaStatus uaStatusFile
+		if err = json.Unmarshal(uaStatusBytes, &uaStatus); err != nil {
+			c.Log.Warnw("unable to unmarshal UA status file", "error", err)
+			return false
+		}
+
+		for _, svc := range uaStatus.Services {
+			if strings.Contains(svc.Name, "esm") && svc.Status == "enabled" {
+				c.Log.Debug("ESM is enabled")
+				return true
+			}
+		}
+
+		c.Log.Debug("no UA service enabled")
+		return false
+	}
+
+	c.Log.Warnw("unable to detect ubuntu ESM support, file not found", "file", procUAStatusFile)
```

Comment:
Should this really be a warn. We're running this check everytime os name is ubuntu. I think this should be an info